hub/core/dataset.py

```diff
@@ -171,11 +172,14 @@ def create_tensor(
 
         Raises:
             TensorAlreadyExistsError: Duplicate tensors are not allowed.
+            InvalidTensorNameError: If `name` is in dataset attributes.
             NotImplementedError: If trying to override `chunk_compression`.
         """
 
         if tensor_exists(name, self.storage):
             raise TensorAlreadyExistsError(name)
+        if name in vars(self).keys():
+            raise InvalidTensorNameError(name)
```

Comment:
Please add a test in `test_api.py` to make sure that this error is actually raised:

```python
def test_invalid_tesnor_name_error(ds):
    with pytest.raises(InvalidTensorNameError):
        ds.create_tensor("meta")
    with pytest.raises(InvalidTensorNameError):
        ds.create_tensor("info")
```